internal/database/metadata/postgres/feature.go

```diff
@@ -102,3 +103,93 @@ func getFeatureByName(ctx context.Context, sqlxCtx metadata.SqlxContext, name st
 
 	return &feature, nil
 }
+
+func listFeature(ctx context.Context, sqlxCtx metadata.SqlxContext, opt metadata.ListFeatureOpt) (types.FeatureList, error) {
+	var (
+		entity   *types.Entity
+		group    *types.Group
+		features types.FeatureList
+		err      error
+	)
+
+	if opt.EntityID != nil {
+		if entity, err = getEntity(ctx, sqlxCtx, *opt.EntityID); err != nil {
+			if errdefs.IsNotFound(err) {
+				return nil, nil
+			}
+			return nil, err
+		}
+	}
+	if opt.GroupID != nil {
+		if group, err = getGroup(ctx, sqlxCtx, *opt.GroupID); err != nil {
+			if errdefs.IsNotFound(err) {
+				return nil, nil
+			}
+			return nil, err
+		}
+	}
```

Comment:
usually, we valid the correctness of argument in callers, like https://github.com/oom-ai/oomstore/blob/main/pkg/oomstore/feature.go#L35-L48, I'm wondering if we need to double check in the db layer?

Reply:
I think the check should be done at this level because y cannot be sure that the caller must have done it. Here it just gets the entity/group based on entityID/groupID and it doesn't seem to be checking either?